### PR TITLE
JAVA-22210 Upgrade dependencies versions for spring-apache-camel module

### PR DESCRIPTION
**Pull request by timis1**
Upgrade to the latest camel version

### CODE REVIEW

File: messaging-modules/spring-apache-camel/src/main/java/com/baeldung/camel/boot/Application.java [hunk 1]
```diff
@@ -21,7 +21,8 @@
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.stereotype.Component;
```
Comment (anastasiosioannidis) [inline messaging-modules/spring-apache-camel/src/main/java/com/baeldung/camel/boot/Application.java:23]: Please undo the identation

File: messaging-modules/spring-apache-camel/pom.xml [hunk 2]
```diff
@@ -77,12 +77,17 @@
...
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-spring-xml</artifactId>
+            <version>${camel.version}</version>
+        </dependency>
     </dependencies>
 
     <properties>
```
Comment (anastasiosioannidis) [inline messaging-modules/spring-apache-camel/pom.xml:87]: Since now 3.12.0 is used, let's delete extra env.camel.version

Comment (timis1) [reply messaging-modules/spring-apache-camel/pom.xml:87]: env.spring.version is used here:
image

File: messaging-modules/spring-apache-camel/src/main/java/com/baeldung/camel/boot/Application.java (hunk 1 above)
Comment (timis1) [reply messaging-modules/spring-apache-camel/src/main/java/com/baeldung/camel/boot/Application.java:23]: DONE

File: messaging-modules/spring-apache-camel/pom.xml (hunk 2 above)
Comment (anastasiosioannidis) [reply messaging-modules/spring-apache-camel/pom.xml:87]: Ok, could we have <version>${spring.version}</version> in line 43?
And also adjust the <properties>

Also replace env.camel.version with camel.version (also adjust <properties>)

Comment (timis1) [reply messaging-modules/spring-apache-camel/pom.xml:87]: Just to rename that property variable ${spring.version}?
${spring.version} is not accessible from parent dependency. It is a spring-boot dependency.

File: messaging-modules/spring-apache-camel/pom.xml [hunk 3]
```diff
@@ -45,7 +45,7 @@
         <dependency>
             <groupId>org.apache.camel</groupId>
             <artifactId>camel-spring-javaconfig</artifactId>
```
Comment (anastasiosioannidis) [inline messaging-modules/spring-apache-camel/pom.xml:47]: It seems we can fully remove the camel-spring-javaconfig, and instead use spring boot
https://stackoverflow.com/questions/73391193/camelconfiguration-deprecated-and-removed

So, remove this and adjust ContentBasedFileRouterConfig class (remove extends and @overrides)

Comment (timis1) [reply messaging-modules/spring-apache-camel/pom.xml:47]: I did the suggested adjustment

**Pull request merged**